### PR TITLE
feat: Phase 2 moving non jest dependent packages

### PR DESCRIPTION
**Pull request by arun3528**
# COMPLETES # [SPARK-513491](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-513491)

## This pull request addresses

This pr covers all the changes done on either the test file or actual code for the packages which can be currently tested in beta branch with all the tests running in mocha 

The Test that are written in mocha can actually be run on jest with little changes, this enables us to move to next branch in phase 

once this is done web client can visualize less changes that need attention , basically web client can move to next branch  next

## by making the following changes

< DESCRIBE YOUR CHANGES >

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

**Review by mkesavan13 — COMMENTED** [review]
- More concerned about the additions that are neither in the beta branch nor in the next branch
- And the tests that are skipped. Are we skipping them in next too?

File: packages/@webex/helper-image/src/orient.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/* eslint-disable complexity */
```
[the remark this answers is not in the document]
Comment (Shreyas281299) [reply packages/@webex/helper-image/src/orient.js:1]: orient method is present in helper-image/src/index.js. In next we have moved it to a new file.

File: packages/@webex/internal-plugin-mercury/test/unit/spec/mercury.js [hunk 2]
```diff
@@ -716,17 +716,99 @@ describe('plugin-mercury', () => {
             .then((wsUrl) => assert.match(wsUrl, /multipleConnections/)));
       });
     });
+  });
+  describe('ping pong latency event is forwarded', () => {
```
Comment (Shreyas281299) [inline packages/@webex/internal-plugin-mercury/test/unit/spec/mercury.js:720]: This tests fails when we run it with all the other tests. https://github.com/webex/webex-js-sdk/commit/b3a5cdf8bc69c5c4ba5439464b85f431d20ad21d - PR in next

File: packages/@webex/helper-image/src/orient.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkesavan13) [reply packages/@webex/helper-image/src/orient.js:1]: But we are not using that new file in this PR.

@Shreyas281299 @arun3528

[the remark this answers is not in the document]
Comment (Shreyas281299) [reply packages/@webex/helper-image/src/orient.js:1]: Yeah exactly. We just need to remove this file because it is not being used anywhere.

File: packages/@webex/internal-plugin-device/test/unit/spec/device.js [hunk 3]
```diff
@@ -12,7 +11,7 @@ describe('plugin-device', () => {
     let webex;
     let device;
 
-    beforeEach('initialize webex with the device plugin', () => {
+    beforeEach(() => {
```
[the remark this answers is not in the document]
Comment (arun3528) [reply packages/@webex/internal-plugin-device/test/unit/spec/device.js:14]: yes

File: packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts [hunk 4]
```diff
@@ -613,7 +613,6 @@ export default class CallDiagnosticMetrics extends StatelessWebexPlugin {
     return this.getErrorPayloadForClientErrorCode({
       clientErrorCode: UNKNOWN_ERROR,
       serviceErrorCode: UNKNOWN_ERROR,
-      payloadOverrides: rawError.payloadOverrides,
```
Comment (marcin-bazyl) [inline packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts:616]: why is this change needed?

File: packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics-batcher.ts [hunk 5]
```diff
@@ -361,8 +361,9 @@ describe('plugin-metrics', () => {
         });
       });
 
+      //TODO: The following two skipped tests needs investigation: https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-485382
```
Comment (marcin-bazyl) [inline packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics-batcher.ts:364]: shouldn't SPARK-485382 be under SPARK-513448? is the plan to fix it before we move webapp to "next"?

File: packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts [hunk 6]
```diff
@@ -1806,27 +1807,6 @@ describe('internal-plugin-metrics', () => {
         });
       });
 
-      it('should override custom properties for an unknown error', () => {
```
Comment (marcin-bazyl) [inline packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts:1809]: why is this test removed?

File: packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts [hunk 7]
```diff
@@ -720,7 +721,7 @@ describe('internal-plugin-metrics', () => {
         ]);
       });
 
-      it('should submit client event successfully with correlationId, webexConferenceIdStr and globalMeetingId', () => {
+      it('should submit client event successfully with correlationId, webexConferenceIdStr and globalMeetingId', async () => {
```
Comment (marcin-bazyl) [inline packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts:724]: why is the `async` needed?

File: packages/@webex/internal-plugin-presence/test/unit/spec/presence.js [hunk 8]
```diff
@@ -7,7 +7,8 @@ import sinon from 'sinon';
 import Presence from '@webex/internal-plugin-presence';
 import MockWebex from '@webex/test-helper-mock-webex';
 
-describe('plugin-presence', () => {
+// Skipping as we have registered it as public plugin and MockWebex will create presence under webex object directly instead of webex.internal
+describe.skip('plugin-presence', () => {
```
Comment (marcin-bazyl) [inline packages/@webex/internal-plugin-presence/test/unit/spec/presence.js:11]: not sure if I understand fully why this is skipped. Should it maybe be completely removed? are these same tests already running against the public plugin?

File: packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts (hunk 4 above)
Comment (Shreyas281299) [reply packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts:616]: This was added recently into beta. We have cherry-picked the commit only today morning. https://github.com/webex/webex-js-sdk/pull/3525. WIll update the branch.

File: packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts [hunk 9]
```diff
@@ -240,24 +240,23 @@ describe('plugin-dss', () => {
...
       });
-
       it('fails with default timeout when mercury does not respond', async () => {
-        const {promise} = await testMakeRequest({
+        return testMakeRequest({
           method: 'lookupDetail',
           resource: '/lookup/orgid/userOrgId/identity/test id/detail',
           params: {id: 'test id'},
           bodyParams: {},
+        }).then(async ({promise}) => {
```
[the remark this answers is not in the document]
Comment (mkesavan13) [reply packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts:249]: > This change doesn't look right. Now if promise doesn't reject, the test will not fail and it should.

I agree. We can add a then block and fail it there. Will do it. 

>  Now if promise doesn't reject, the test will not fail and it should. Also the clock.tickAsync() is now no longer called before the check for the promise being rejected, it may be called before or after, which will probably make the test more flakey.

So, a little context for you here:

Earlier, the mocha tests were letting the Promises that were rejected go without any breakage. So, the code reached the assertion for checking isRejected. With new tooling and stricter checks, the test breaks as soon as the promise rejects because there's no catch mechanism. 

In the current code in beta,

1. We create a promise
2. We tick the clock to 6s and as soon as that happens, the promise rejects
3. When the promise rejects, we check the assertion for `isRejected`

In the newer code from next,

1. We create a promise
2. Write a catch block for When the promise would fail (The code won't execute the error block before tickAsync is executed)
3. We do `clock.tickAsync()` which makes the promise reject
4. When this happens, it goes into the catch block and checks if it has received the expected error message

So, to address your concern, we are not checking if the promise was rejected before `clock.tickAsync()` rather just added a catch mechanism for whenever it fails. Adding a then mechanism if it passes and failing the test would make this test whole I believe.

File: packages/@webex/plugin-messages/src/messages.js [hunk 10]
```diff
@@ -160,6 +160,83 @@ const Messages = WebexPlugin.extend({
...
+   *     assert(message.personId);
+   *     assert(message.personEmail);
+   *     assert(message.roomId);
+   *     assert(message.created);
+   *     return 'success';
+   *   });
+   *   // => success
+   */
+  update(message, altMessage) {
+    const id = message.id || message;
```
Comment (Shreyas281299) [inline packages/@webex/plugin-messages/src/messages.js:198]: @arun3528 Do we need these changes in beta and next?
https://github.com/webex/webex-js-sdk/pull/2708

File: packages/@webex/internal-plugin-mercury/test/unit/spec/mercury.js (hunk 2 above)
Comment (Shreyas281299) [reply packages/@webex/internal-plugin-mercury/test/unit/spec/mercury.js:749]: Since we have changed mercury to use mocha, this change is not required anymore.

File: packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics-batcher.ts (hunk 5 above)
Comment (Shreyas281299) [reply packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics-batcher.ts:364]: Since we have changed metrics to use mocha these test are passing. When we move metrics to use jest these tests needs to be looked into.

File: packages/@webex/test-helper-mock-webex/src/index.js [hunk 11]
```diff
@@ -25,6 +25,7 @@ const nonInternalPlugins = [
   'teams',
   'teamMemberships',
   'webhooks',
+  'presence',
```
Comment (Shreyas281299) [inline packages/@webex/test-helper-mock-webex/src/index.js:28]: I dont think we need this in beta. @mkesavan13 can you please confirm this.

File: packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts [hunk 12]
```diff
@@ -1156,8 +1155,8 @@ describe('plugin-dss', () => {
         expect(Batcher.prototype.request.getCall(1).args).to.deep.equal(['id2']);
         expect(result).to.equal(response2);
       });
-
-      it('fails fails when mercury does not respond, later batches can still pass ok', async () => {
+      // TODO
+      it.skip('fails fails when mercury does not respond, later batches can still pass ok', async () => {
```
[the remark this answers is not in the document]
Comment (Shreyas281299) [reply packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts:1159]: As Kesava explained the approach that we have taken to make the tests runs. The issue with this test is that we have two requests going out, just using .then and .catch we cannot achieve the same behaviour. Added JIRA

File: packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts (hunk 7 above)
Comment (Shreyas281299) [reply packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts:724]: Not required. Removed

File: packages/@webex/test-helper-mock-webex/src/index.js (hunk 11 above)
Comment (mkesavan13) [reply packages/@webex/test-helper-mock-webex/src/index.js:28]: Yes. This is also the presence plugin change. So, we can revert this too.

File: packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts (hunk 6 above)
Comment (Shreyas281299) [reply packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts:1809]: Same as above,
`This was added recently into beta. We have cherry-picked the commit only today morning. https://github.com/webex/webex-js-sdk/pull/3525. WIll update the branch.`

File: packages/@webex/test-helper-mock-webex/src/index.js (hunk 11 above)
Comment (mkesavan13) [reply packages/@webex/test-helper-mock-webex/src/index.js]: Reverted

File: packages/@webex/plugin-messages/src/messages.js (hunk 10 above)
Comment (arun3528) [reply packages/@webex/plugin-messages/src/messages.js:198]: yes i remember someone added it , its been a while , why what happened ?

File: packages/@webex/internal-plugin-presence/test/unit/spec/presence.js (hunk 8 above)
Comment (arun3528) [reply packages/@webex/internal-plugin-presence/test/unit/spec/presence.js]: you are correct we dont need to move it to beta , just skipped it as we had a public one

File: packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (mkesavan13) [reply packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts:249]: Ah. Clever one. Thanks, Marcin. We can do this. And also see if we can follow a similar approach to the above-skipped test.

> Yes, agreed, that would be fine, but I don't see promise.then() being done in this PR.

About this, we were waiting for your signal, if my explanation was convincing to you so that we can make the change. Now that we go with the old approach and add just a line, we won't need it.

File: packages/@webex/internal-plugin-presence/test/unit/spec/presence-worker.js [hunk 13]
```diff
@@ -26,13 +26,35 @@ describe('presence-worker', () => {
...
+        }
+        assert.equal(err?.message, 'Must initialize Presence Worker with webex!'); 
+
+      });
+
     });
 
-    describe('#enqueue()', () => {
+    // This selection of tests fail due to `webex-core`'s batcher config being missing.
+    describe.skip('#enqueue()', () => {
```
Comment (marcin-bazyl) [inline packages/@webex/internal-plugin-presence/test/unit/spec/presence-worker.js:57]: when is this test going to be re-enabled? do we have a jira?

File: packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (mkesavan13) [reply packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts:249]: PR to revert this: https://github.com/webex/webex-js-sdk/pull/3564

File: packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts (hunk 12 above)
[the remark this answers is not in the document]
Comment (mkesavan13) [reply packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts:1159]: We'll skip this only in the next branch. 

This PR doesn't need this change. So, will revert from the PR

[the remark this answers is not in the document]
Comment (mkesavan13) [reply packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts]: Reverted.

File: packages/@webex/internal-plugin-presence/test/unit/spec/presence-worker.js (hunk 13 above)
Comment (mkesavan13) [reply packages/@webex/internal-plugin-presence/test/unit/spec/presence-worker.js:57]: @arun3528 - Do you want to take this?

Comment (arun3528) [reply packages/@webex/internal-plugin-presence/test/unit/spec/presence-worker.js:57]: @mkesavan13  if I am not wrong this was failing in beta too , but let's create a ticket to fix this for presence public ga. For now I think it's OK as it does not have source code change

File: packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (mkesavan13) [reply packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts]: PR merged and cherry-picked into this PR. Please check

File: packages/@webex/plugin-messages/src/messages.js (hunk 10 above)
Comment (mkesavan13) [reply packages/@webex/plugin-messages/src/messages.js:198]: @Shreyas281299 - Can we resolve this?

File: packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (marcin-bazyl) [reply packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts]: I see it's fixed for this test, but there are at least 4 other tests with the same issue in this file that are not fixed

File: packages/@webex/internal-plugin-presence/test/unit/spec/presence-worker.js (hunk 13 above)
Comment (marcin-bazyl) [reply packages/@webex/internal-plugin-presence/test/unit/spec/presence-worker.js:57]: sorry, I'm confused. All I see in this PR diff is that  a `skip` is added. Surely it was not failing in beta as we have PRs landing on beta every day with tests passing without that skip. If a skip is really needed, raise a jira to fix this, put it under SPARK-513448 and add a comment with that jira in here, please

File: packages/@webex/plugin-messages/src/messages.js (hunk 10 above)
Comment (Shreyas281299) [reply packages/@webex/plugin-messages/src/messages.js:198]: Yes. Resolving

**Review by marcin-bazyl — APPROVED** [review]

**Comment by arun3528** [comment]
@mkesavan13 @Shreyas281299  @sreenara  merging it , knowing the feature plugin is broken in beta as well
